src/nccl_ofi_net.c

```diff
@@ -1142,6 +1221,44 @@ static ncclResult_t ofi_init(ncclDebugLogger_t logFunction)
 			       ofi_info_list->fabric_attr->prov_name);
 	}
 
+	/* Check if provider uses remote virtual addressing */
+	if (ofi_info_list->domain_attr->mr_mode & FI_MR_VIRT_ADDR) {
+		NCCL_OFI_TRACE(NCCL_INIT | NCCL_NET, "Provider %s uses remote virtual addressing",
+			       ofi_info_list->fabric_attr->prov_name);
+		virt_addr_mr = true;
```

Comment:
Thank you for catching this bug :) Do you mind separating this out in a different commit?

Reply:
Are you thinking of a separate PR to add offset-based addressing? Or a separate MR to check this bit and error if it's not set? I don't think I have a good way to test a PR for the former outside of this full PR. The latter is certainly doable if you would find it useful.

Reply:
Ahh you beat me to it :) Just a separate PR as this is a different logic that MR bits and keys patch.

Reply:
I could put something together for this, but unfortunately I don't have any provider where I could verify that everything is working correctly. Slingshot provider requires the other changes in this PR.

Reply:
That's okay! We can test the original behavior on our systems.

Reply:
Is there a provider that wants to use offset-based addressing (`! FI_MR_VIRT_ADDR`) with the existing aws-ofi-nccl master branch? I'm trying to understand why to split off these changes based on my understanding that no provider has required this support until now and that we'll also be merging these changes soon along with the rest of this PR. As mentioned earlier, I can make a best effort to do this, but I don't have any way to test that the branch I produce works correctly and would need someone with such a provider to take on the work of merging such a branch. Given the above, I had suggested an error check for `FI_MR_VIRT_ADDR` which is something that I can generate and check with reasonable confidence. Please let me know how you'd like to proceed. Wish there was more I could do to help, but I'm limited by the providers available to me.